hal/halx86/generic/halinit.c

```diff
@@ -12,6 +12,14 @@
 #define NDEBUG
 #include <debug.h>
 
+#ifdef _WIN64
+#define InterlockedBitTestAndSetPointer(ptr, val) \
```

Comment:
Would you prefer to have it somewhere (where?) in SDK, instead of duplicating it each time it is needed?

Reply:
Since it's clear you spied on Mattermost to see what I was mentioning to @tkreuzer :trollface: , please use instead the name he suggested:
![image](https://github.com/reactos/reactos/assets/1969829/f1b1a534-5ed8-4a63-9524-68c7c923891d)

Reply:
> Since it's clear you spied on Mattermost to see what I was mentioning to @tkreuzer :trollface: ,

?? I have not been on Mattermost since ages!

> please use instead the name he suggested

I "copied" this code from
https://github.com/reactos/reactos/blob/a79bd4d0cca7a377153007085cb661edc95306ef/sdk/lib/rtl/srw.c#L24

I would be fine to use 'Affinity' name,
but do you really want multiple times the same code, with different names even?
Note:
That will also be needed for "PushLock" and whatnot...